File: array linked list.py

class Queue:

	def __init__(self, c):
		
		self.queue = []
		self.front = self.rear = 0
		self.capacity = c

	def queueEnqueue(self, data):

		if(self.capacity == self.rear):
			print("\nQueue is full")

		else:
			self.queue.append(data)
			self.rear += 1

	def queueDequeue(self):

		if(self.front == self.rear):
			print("Queue is empty")

		else:
			x = self.queue.pop(0)
			self.rear -= 1

	def queueDisplay(self):
		
		if(self.front == self.rear):
			print("\nQueue is Empty")

		for i in self.queue:
			print(i, "<--", end = '')
	
	
	def queueFront(self):
		
		if(self.front == self.rear):
			print("\nQueue is Empty")

		print("\nFront Element is:",
			self.queue[self.front])

if __name__=='__main__':

	q = Queue(4)

	q.queueDisplay()

	q.queueEnqueue(44)
	q.queueEnqueue(55)
	q.queueEnqueue(66)
	q.queueEnqueue(77)

	q.queueDisplay()

	q.queueEnqueue(66)

	q.queueDisplay()

	q.queueDequeue()
	q.queueDequeue()
	print("\n\nafter two node deletion we have \n")

	q.queueDisplay()

	q.queueFront()

File: array linked list.cpp

#include <bits/stdc++.h>
using namespace std;

struct Queue {
	int front, rear, capacity;
	int* queue;
	Queue(int c)
	{
		front = rear = 0;
		capacity = c;
		queue = new int;
	}

	~Queue() { delete[] queue; }


	void queueEnqueue(int data)
	{
		
		if (capacity == rear) {
			printf("\nQueue is full\n");
			return;
		}

	
		else {
			queue[rear] = data;
			rear++;
		}
		return;
	}

	
	void queueDequeue()
	{
	
		if (front == rear) {
			printf("\nQueue is empty\n");
			return;
		}

		
		else {
			for (int i = 0; i < rear - 1; i++) {
				queue[i] = queue[i + 1];
			}

			
			rear--;
		}
		return;
	}

	
	void queueDisplay()
	{
		int i;
		if (front == rear) {
			printf("\nQueue is Empty\n");
			return;
		}

		
		for (i = front; i < rear; i++) {
			printf(" %d <-- ", queue[i]);
		}
		return;
	}

	
	void queueFront()
	{
		if (front == rear) {
			printf("\nQueue is Empty\n");
			return;
		}
		printf("\nFront Element is: %d", queue[front]);
		return;
	}
};


int main(void)
{
	Queue q(4);


	q.queueDisplay();

	q.queueEnqueue(21);
	q.queueEnqueue(31);
	q.queueEnqueue(41);
	q.queueEnqueue(51);

	q.queueDisplay();

	q.queueEnqueue(61);

	q.queueDisplay();

	q.queueDequeue();
	q.queueDequeue();

	printf("\n\nafter two node deletion\n\n");

	q.queueDisplay();

	q.queueFront();

	return 0;
}
